mifosng-provider/src/main/java/org/mifosplatform/accounting/provisioning/service/ProvisioningEntriesWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -0,0 +1,216 @@
+
+import com.google.gson.JsonObject;
+
+@Service
+public class ProvisioningEntriesWritePlatformServiceJpaRepositoryImpl implements ProvisioningEntriesWritePlatformService {
+
+    private final static Logger logger = LoggerFactory.getLogger(ProvisioningEntriesWritePlatformServiceJpaRepositoryImpl.class);
+
+    private final ProvisioningEntriesReadPlatformService provisioningEntriesReadPlatformService;
+    private final LoanProductRepository loanProductRepository;
+    private final GLAccountRepository glAccountRepository;
+    private final OfficeRepository officeRepository;
+    private final ProvisioningCategoryRepository provisioningCategoryRepository;
+    private final PlatformSecurityContext platformSecurityContext;
+    private final ProvisioningEntryRepository provisioningEntryRepository;
+    private final JournalEntryWritePlatformService journalEntryWritePlatformService;
+    private final ProvisioningEntriesDefinitionJsonDeserializer fromApiJsonDeserializer;
+    private final FromJsonHelper fromApiJsonHelper;
+
+    @Autowired
+    public ProvisioningEntriesWritePlatformServiceJpaRepositoryImpl(
+            final ProvisioningEntriesReadPlatformService provisioningEntriesReadPlatformService,
+            final LoanProductRepository loanProductRepository, final GLAccountRepository glAccountRepository,
+            final OfficeRepository officeRepository, final ProvisioningCategoryRepository provisioningCategoryRepository,
+            final PlatformSecurityContext platformSecurityContext, final ProvisioningEntryRepository provisioningEntryRepository,
+            final JournalEntryWritePlatformService journalEntryWritePlatformService,
+            final ProvisioningEntriesDefinitionJsonDeserializer fromApiJsonDeserializer, final FromJsonHelper fromApiJsonHelper) {
+        this.provisioningEntriesReadPlatformService = provisioningEntriesReadPlatformService;
+        this.loanProductRepository = loanProductRepository;
+        this.glAccountRepository = glAccountRepository;
+        this.officeRepository = officeRepository;
+        this.provisioningCategoryRepository = provisioningCategoryRepository;
+        this.platformSecurityContext = platformSecurityContext;
+        this.provisioningEntryRepository = provisioningEntryRepository;
+        this.journalEntryWritePlatformService = journalEntryWritePlatformService;
+        this.fromApiJsonDeserializer = fromApiJsonDeserializer;
+        this.fromApiJsonHelper = fromApiJsonHelper;
+    }
+
+    @Override
+    public CommandProcessingResult createProvisioningJournalEntries(Long provisioningEntryId, JsonCommand command) {
+        ProvisioningEntry requestedEntry = this.provisioningEntryRepository.findOne(provisioningEntryId);
+        if (requestedEntry == null) { throw new ProvisioningEntryNotfoundException(provisioningEntryId); }
+
+        ProvisioningEntry existingEntry = this.provisioningEntryRepository.findExistingProvisioningEntryWithJournalEntries();
```

Comment:
existingEntry  is not required here. you can get latest entry date and validate here